File: 0053-maximum-subarray/0053-maximum-subarray.cs
public class Solution {
    public int MaxSubArray(int[] nums) {
        int maxSum = int.MinValue, currSum = 0;
        foreach(var num in nums){
            currSum = Math.Max(currSum + num, num);
            maxSum = Math.Max(maxSum, currSum);
        }
        
        return maxSum;
    }
}

File: 0053-maximum-subarray/0053-maximum-subarray.java
class Solution {
    public int maxSubArray(int[] nums) {
        int maxSum = Integer.MIN_VALUE, currSum = 0;
        for(var num : nums){
            currSum = Math.max(currSum + num, num);
            maxSum = Math.max(maxSum, currSum);
        }
        
        return maxSum;
    }
}